src/main/scala/diplomacy/Nodes.scala

```diff
@@ -545,15 +595,27 @@ class MixedAdapterNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
   override def description = "adapter"
   protected[diplomacy] override def flexibleArityDirection = true
   protected[diplomacy] def resolveStar(iKnown: Int, oKnown: Int, iStars: Int, oStars: Int): (Int, Int) = {
-    require (oStars + iStars <= 1, s"$context appears left of a :*= $iStars times and right of a :=* $oStars times; at most once is allowed")
+    require (oStars + iStars <= 1,
+      s"""$context
+         |$bindingInfo
+         |appears left of a :*= $iStars times and right of a :=* $oStars times; at most once is allowed""".stripMargin)
     if (oStars > 0) {
-      require (iKnown >= oKnown, s"$context has $oKnown outputs and $iKnown inputs; cannot assign ${iKnown-oKnown} edges to resolve :=*")
+      require (iKnown >= oKnown,
+        s"""$context
+           |$bindingInfo
+           |has $oKnown outputs and $iKnown inputs; cannot assign ${iKnown-oKnown} edges to resolve :=*""".stripMargin)
       (0, iKnown - oKnown)
     } else if (iStars > 0) {
-      require (oKnown >= iKnown, s"$context has $oKnown outputs and $iKnown inputs; cannot assign ${oKnown-iKnown} edges to resolve :*=")
+      require (oKnown >= iKnown,
+        s"""$context
+           |$bindingInfo
+           |has $oKnown outputs and $iKnown inputs; cannot assign ${oKnown-iKnown} edges to resolve :*=""".stripMargin)
       (oKnown - iKnown, 0)
     } else {
-      require (oKnown == iKnown, s"$context has $oKnown outputs and $iKnown inputs; these do not match")
+      require (oKnown == iKnown,
+        s"""$context
+           |$bindingInfo
+           |has $oKnown outputs and $iKnown inputs; these do not match""".stripMargin)
```

Comment:
`The following node appears outwards of a := $iKnown times and inwards of a := $oKnown times. The number of bindings on both sides of the node must match.`